Bypassing the error

I'm using the source-based Rust coverage but occasionally forgot to install an `llvm-tools-preview` component.
In CI I got the green job with no coverage, but in fact, one error passed unnoticed on `grcov` side:

<details>

```
test result: ok. 4 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 1.24s
$ grcov . --binary-path ./target/debug/ --source-dir . --output-type lcov --llvm --branch --ignore-not-existing --ignore "/*" --ignore "tests/*" --output-path lcov-w-branch.info
16:34:00 [ERROR] Error while executing llvm tools: Failed to execute "/usr/local/rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-profdata" ...

# many file names

...
"/tmp/.tmps2z62Q/crates/lang/llvmcoveragedata-6052-17221853374265077399_0_1.profraw" "-o" "/tmp/.tmps2z62Q/0/grcov.profdata"
No such file or directory (os error 2)

$ rust-covfix lcov-w-branch.info --output lcov-w-branch-fixed.info
Coverages are fixed successfully!
  line:   NaN% (0 of 0 lines)    => NaN% (0 of 0 lines)
  branch: NaN% (0 of 0 branches) => NaN% (0 of 0 branches)
```

</details>

I didn't find a flag to specify an exit code.

CC: https://github.com/mozilla/grcov/issues/656 https://github.com/mozilla/grcov/issues/555

--- a/src/llvm_tools.rs
+++ b/src/llvm_tools.rs
@@ -83,8 +83,8 @@ fn get_profdata_path() -> Result<PathBuf, String> {
     let path = Tool::Profdata.path().map_err(|x| x.to_string())?;
     if !path.exists() {
         Err(String::from(
-            "We couldn't find llvm-profdata. \
-             Do you have the llvm-tools component installed?"
+            "We couldn't find llvm-profdata. Try installing the llvm-tools \
+            component with `rustup component add llvm-tools-preview`."
         ))
     } else {
         Ok(path)

--- a/src/llvm_tools.rs
+++ b/src/llvm_tools.rs
@@ -38,7 +38,8 @@ pub fn profraws_to_lcov(
         profdata_path.as_ref(),
     ];
     args.splice(2..2, profraw_paths.iter().map(PathBuf::as_ref));
-    run(&Tool::Profdata.path().unwrap(), &args)?;
+
+    let path = get_profdata_path().and_then(|p| run(&p, &args))?;
 
     let binaries = if binary_path.is_file() {
         vec![binary_path.to_owned()]
@@ -76,6 +77,18 @@ pub fn profraws_to_lcov(
     }
 
     Ok(results)
+}
+
+fn get_profdata_path() -> Result<PathBuf, String> {
+    let path = Tool::Profdata.path().map_err(|x| x.to_string())?;
+    if !path.exists() {
+        Err(String::from(
+            "We couldn't find llvm-profdata. \
+             Do you have the llvm-tools component installed?"
+        ))
+    } else {
+        Ok(path)
+    }
 }
 
 #[cfg(test)]

--- a/src/llvm_tools.rs
+++ b/src/llvm_tools.rs
@@ -39,7 +39,7 @@ pub fn profraws_to_lcov(
     ];
     args.splice(2..2, profraw_paths.iter().map(PathBuf::as_ref));
 
-    let path = get_profdata_path().and_then(|p| run(&p, &args))?;
+    get_profdata_path().and_then(|p| run(&p, &args))?;
 
     let binaries = if binary_path.is_file() {
         vec![binary_path.to_owned()]

